src/artifact_providers/base.ts

```diff
@@ -61,14 +62,47 @@ export interface LocalArtifact extends AbstractArtifact {
 }
 
 /**
- * Fitlering options for artifacts
+ * Raw filtering options for artifacts
  */
-export interface FilterOptions {
+export interface RawFilterOptions {
+  /** Include files that match this pattern */
+  includeNames?: RegExp | string;
+  /** Exclude files that match this pattern */
+  excludeNames?: RegExp | string;
+}
+
+/**
+ * Parsed filtering options for artifacts
+ */
+export interface ParsedFilterOptions {
   /** Include files that match this regexp */
   includeNames?: RegExp;
   /** Exclude files that match this regexp */
   excludeNames?: RegExp;
 }
+
+/**
+ * Returns parsed the given raw filters.
+ */
+export function parseFilterOptions(
+  rawFilters: RawFilterOptions
+): ParsedFilterOptions {
+  const parsedFilters: ParsedFilterOptions = {};
+  if (typeof rawFilters.includeNames !== 'undefined') {
```

Comment:
Do we need the undefined check if the properties (ex. `includesNames`) for `ParsedFilterOptions` are typed as optional?